storage/lmdb/src/initial_map_size.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::memsize::MemSize;
+
+pub struct InitialMapSize {
+    initial_map_size: MemSize,
+}
+
+impl InitialMapSize {
+    pub fn into_memsize(&self) -> MemSize {
+        self.initial_map_size
+    }
+}
+
+impl Default for InitialMapSize {
+    fn default() -> Self {
+        Self {
+            initial_map_size: MemSize::from_bytes(0),
```

Comment:
Is this correct to pass 0 as the initial size? I think to get the original behaviour, `InitialMapSize` would contain and `Option<MemSize>` and default would be `None`. That way we are left with whatever LMDB does by default when `InitialMapSize::default` is used. I don't know whether 0 behaves the same or not.

Reply:
I'll go with None. I think that's more reasonable.

Zero in lmdb means "nothing will happen"... it'll basically be ignored. But I guess you're right it's up to lmdb.

Reply:
Heads up on the ugliness because of this... you'll see why I chose this.